test/test_helper.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+"""
+Copyright 2016 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import platform
+import os
+import mbed_lstools
+
+
+class Helper(object):
+    """
+    Helper class for all tests
+    """
+    def __init__(self, platform_name, allowed_files):
+        self.platform_name = platform_name
+        self.allowed_files = []
+        self.allowed_files.extend(allowed_files)
+
+    def clear(self):
+        """
+        clear Files should not be put into mount point
+        """
+        if self.platform_name:
+            targets = self._get_targets()
+
+            # Clear files other than allowed
+            for target in targets:
+                mount_point = target.get('mount_point')
+                if mount_point:
+                    for bad_file in os.listdir(mount_point):
+                        if bad_file not in self.allowed_files:
+                            self.del_file(mount_point, bad_file)
+
+    def _get_targets(self):
+        """
+        Return targets matching passed platform name
+        """
+        mbeds = mbed_lstools.create()
+        targets = mbeds.list_mbeds()
+        selected_targets = []
+
+        if targets:
+            for target in targets:
+                if target['platform_name'] == self.platform_name:
+                    selected_targets.append(target)
+
+        return selected_targets
+
+
+    @staticmethod
+    def del_file(file_path, file_name):
+        """
+        :param file_path: file path
+        :param file_name: file name of file to be deleted
+        :return:
+        """
+        if os.path.isfile(os.path.join(file_path, file_name)):
+            if platform.system() == 'Windows':
+                os.system('del /F %s' % os.path.join(file_path, file_name))
```

Comment:
Why not use os.remove?

Reply:
now it's 6 mins, no big difference with os.remove

Reply:
now we're facing issue of deleting windows "System Volume Information"
@juhhov  suggested to just delete FAIL.txt and ASSERT.TXT
@jupe  how do think?
````
10:10:13 [windows]     Helper(platform_name='K64F', allowed_files=['DETAILS.TXT', 'MBED.HTM']).clear()
10:10:13 [windows]   File "C:\Jenkins\ws\mbed-flasher_PR-70-QZD4OYH22FQ2ZZO2T2SBCPXY6APY2\mbed-flasher\test\test_helper.py", line 44, in clear
10:10:13 [windows]     os.remove(os.path.join(mount_point, bad_file))
10:10:13 [windows] PermissionError: [WinError 5] Access is denied: 'D:System Volume Information'
```

Reply:
seems that bad_file points to invalid place... something wrong in code side I think

Reply:
```C:\>dir /a:hd D:
 Volume in drive D is DAPLINK
 Volume Serial Number is 2702-1974
 Directory of D:\
13.10.2017  12.08    <DIR>          System Volume Information
               0 File(s)              0 bytes
               1 Dir(s)      67 088 384 bytes free
```

Reply:
This should be the correct place. windows seems create this file for each USB connected device

Reply:
yes, you're right, this is a bug, fixed now. :)